src/index.tsx

```diff
@@ -415,13 +415,15 @@ const Intercom: IntercomType = {
       IntercomEventEmitter.startEventListener();
     const eventEmitter = new NativeEventEmitter(IntercomEventEmitter);
     const listener = eventEmitter.addListener(event, callback);
-    listener.remove = () => {
-      event === IntercomEvents.IntercomUnreadCountDidChange &&
-        Platform.OS === 'android' &&
-        IntercomEventEmitter.removeEventListener();
-      listener.remove();
+    return {
+      ...listener,
+      remove: () => {
+        event === IntercomEvents.IntercomUnreadCountDidChange &&
+          Platform.OS === 'android' &&
+          IntercomEventEmitter.removeEventListener();
+        listener.remove();
+      },
     };
```

Comment:
One option that shouldn't break the ts return type is to store the "original" remove function beforehand:
```suggestion
    const originalRemove = listener.remove;
    listener.remove = () => {
      event === IntercomEvents.IntercomUnreadCountDidChange &&
        Platform.OS === 'android' &&
        IntercomEventEmitter.removeEventListener();
      originalRemove();
    };
    return listener;
```

Reply:
Yep, this works. Thanks @eltonvs